src/kixi/eventlog/web_server.clj

```diff
@@ -1,54 +1,59 @@
             [kixi.event.topic :refer [publish]]
             [org.httpkit.server :as http-kit]
-            [ring.middleware.defaults :refer [wrap-defaults api-defaults]]))
+            [ring.middleware.defaults :refer [wrap-defaults api-defaults]]
+            [buddy.sign.jwt :as jwt]
+            [kixi.eventlog.authz :as authz]))
 
-(defn status-routes []
-  (routes
-   (GET "/_elb_status" []  "ALL GOOD")
-   (POST "/_elb_status" []  "ALL GOOD")))
+(defroutes status-routes
+  (GET "/_elb_status" []  "ALL GOOD")
+  (POST "/_elb_status" []  "ALL GOOD"))
 
 (defn path-for-topic [topic-name]
-  
   (let [name (if-let [[name _] (next (re-matches #"(\w+?)(\d+)" topic-name))]
                name
                topic-name)
         path (str "/" (str/replace name #"_" "/"))]
     (log/infof "Path for topic %s is %s" topic-name path)
     path))
 
-(defn topic-routes [producer topic-names]
+(defn topic-routes [component]
   (let [topic-routes (map #(POST (path-for-topic %) []
-                                 (index-resource (partial publish producer %))) topic-names)]
+                                 (index-resource (partial publish (:producer component) %))) (-> component :topics :topic-defs keys))]
     (apply routes topic-routes)))
 
-(defrecord WebServer [opts]
+(defn app
+  [component authentication? auth]
+  (routes
+   status-routes
+   (wrap-routes  (topic-routes component) (authz/maybe-wrap-authentication authentication? auth))
+   (not-found {:headers {"Content-Type" "application/json"}
+               :body    "{\"error\": \"No Such Endpoint\"}"})))
+
+(defrecord WebServer [opts authentication? auth]
   component/Lifecycle
   (start [this]
     (log/info "Starting Webserver")
-    (let [routes                       (routes
-                                        (topic-routes (:producer this) (-> this :topics :topic-defs keys))
-                                        (status-routes)
-                                        (not-found {:headers {"Content-Type" "application/json"}
-                                                    :body    "{\"error\": \"No Such Endpoint\"}"}))]
-      (let [server (http-kit/run-server (wrap-defaults
-                                         routes
-                                         api-defaults)
-                                        opts)]
-        (assoc this ::server server))))
+    (let [server (http-kit/run-server (wrap-defaults
+                                       (app this authentication? auth)
+                                       api-defaults)
+                                      opts)]
+      (assoc this ::server server)))
   (stop [this]
     (log/info "Stopping Webserver")
     (when-let [close-fn (::server this)]
       (close-fn))))
 
-(defn new-server []
+(defn new-server [authentication? auth]
   (->WebServer {:verbose? true
                 :port 8080
                 :max-body
-                (* 16 1024 1024)}))
+                (* 16 1024 1024)}
```

Comment:
this is the Max Message Size. I think we should factor this out to MAX_MESSAGE_SIZE somewhere else.